xarray/tests/test_backends.py

```diff
@@ -2091,6 +2091,68 @@ def test_chunks(self):
                 ex = expected.sel(band=1).mean(dim='x')
                 assert_allclose(ac, ex)
 
+    def test_ENVI_tags(self):
+        rasterio = pytest.importorskip('rasterio', minversion='1.0a')
+        from rasterio.transform import from_origin
+
+        # Create an ENVI file with some tags in the ENVI namespace
+        with create_tmp_file(suffix='.dat') as tmp_file:
+            # data
+            nx, ny, nz = 4, 3, 3
+            data = np.arange(nx*ny*nz,
+                             dtype=rasterio.float32).reshape(nz, ny, nx)
+            transform = from_origin(5000, 80000, 1000, 2000.)
+            with rasterio.open(
+                    tmp_file, 'w',
+                    driver='ENVI', height=ny, width=nx, count=nz,
+                    crs={'units': 'm', 'no_defs': True, 'ellps': 'WGS84',
+                         'proj': 'utm', 'zone': 18},
+                    transform=transform,
+                    dtype=rasterio.float32) as s:
+                s.update_tags(
+                        ns='ENVI',
+                        description='{Tagged file}',
+                        wavelength='{123.000000, 234.234000, 345.345678}',
+                        fwhm='{1.000000, 0.234000, 0.000345}',
+                        )
+                s.write(data)
+                dx, dy = s.res[0], -s.res[1]
+
+            # Tests
+            expected = DataArray(data, dims=('band', 'y', 'x'),
+                                 coords={
+                                     'band': [1, 2, 3],
+                                     'y': -np.arange(ny) * 2000 + 80000 + dy/2,
+                                     'x': np.arange(nx) * 1000 + 5000 + dx/2,
+                                     'wavelength': (
+                                         'band',
+                                         np.array([123, 234.234, 345.345678]),
+                                         ),
+                                     'fwhm': (
+                                         'band',
+                                         np.array([1, 0.234, 0.000345]),
+                                         ),
+                                    },
+                                 )
```

Comment:
nit: can you collapse this constructor a bit. Here (and elsewhere in this PR) the extra lines/brackets are just a bit extraneous and diverge from the style elsewhere in the project.